### DISCUSSION TITLE
RLMatrix with Stride, Assembly problem

### DISCUSSION THREAD

**Discussion by delustra** [asieradzk/RL_Matrix#10] [General]
Hey, trying to use RLMatrix with Stride 3d, based on the godot examples. But it causes errors, when I include the package into the build. Some assembly stuff is breaking and doesn't let the engine editor work properly. I see the comment in RLMatrixGodotTrainingManager.cs with a workaround. Could you try to fix it? I think you are right in the comment, it's binary serialiser related (it's deprecated in net 8.0).

**Comment by asieradzk** [comment]
Hi @delustra Thanks for trying RLMatrix with Stride - you got there before me. :)

it's binary serialiser related (it's deprecated in net 8.0).
-Yep. I am currently working on major update & re-work that will also include replacing binary serialiser with something else. Feel free to make suggestions what to use to serialise replay buffer. I'm thinking about this:
https://github.com/realm/realm-dotnet

It should hopefully be out this week but for now you can try to lower project version to net 6.0

**Reply by delustra** [reply]
It actually acts somewhat strange. If I use RLMatrix from code, build the game, it works - no problems with compiling. The problem is an editor - as soon as I add nuget package to the project, the editor starts to generate crash reports and acts funny. When I remove the lib - all fine again. Hard to say, what's the problem there.
